### PR TITLE
[Relay][Op]BroadcastToLike CollapseSumLike

### PR DESCRIPTION
**Pull request by MarisaKirisame**
Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from others in the community.

### CODE REVIEW

File: src/relay/op/tensor/transform.cc [hunk 1]
```diff
@@ -498,5 +498,66 @@ and type as the input array.
...
 .add_type_rel("FullLike", FullLikeRel);
 
+// Have no idea how to assert the constraint.
+// CollapseSumLike: <A, B> -> B where BroadCast(A, B) = A
+bool CollapseSumLikeRel(const Array<Type>& types,
+                        int num_inputs,
+                        const Attrs& attrs,
+                        const TypeReporter& reporter) {
+  CHECK_EQ(types.size(), 3);
+  reporter->Assign(types[2], types[1]);
```
Comment (slyubomirsky) [inline src/relay/op/tensor/transform.cc]: Maybe see ReduceShapeImpl in https://github.com/dmlc/tvm/blob/master/nnvm/src/top/tensor/reduce.cc could provide guidance on how the axis params are supposed to work? (Strangely, `collapse_sum` itself doesn't use the reduction.)

Comment (MarisaKirisame) [reply src/relay/op/tensor/transform.cc]: I am confused as well - I originally thought I need reduce_shape (which was at another pr), so at the mean time I can get this up.

**Comment by MarisaKirisame** [comment]
@tqchen can you give some review?

**Comment by tqchen** [comment]
please rebase against master after #1934 to make use of the newly introduced API and add test-case to make sure text format works.

**Comment by MarisaKirisame** [comment]
@tqchen I had rebased.

**Comment by tqchen** [comment]
The changes in the submodule seems to be unintentional and should not be included in this PR

**Pull request merged**